src/python/pants/backend/go/util_rules/build_pkg_target.py

```diff
@@ -300,16 +351,34 @@ async def setup_build_go_package_target_request(
         if assembler_flags_field and assembler_flags_field.value:
             pkg_specific_assembler_flags = assembler_flags_field.value
 
-    all_direct_dependencies = await Get(Targets, DependenciesRequest(target[Dependencies]))
+    def make_stdlib_dep(import_path: str) -> Address:
+        return DEFAULT_GO_SDK_ADDR.create_generated(import_path)
+
+    extra_dependencies_addrs = []
+    if cgo_files and (not is_stdlib or import_path != "runtime/cgo"):
+        extra_dependencies_addrs.append(make_stdlib_dep("runtime/cgo"))
+    if cgo_files and (
+        not is_stdlib
+        or import_path not in ("runtime/cgo", "runtime/race", "runtime/msan", "runtime/asan")
+    ):
+        extra_dependencies_addrs.append(make_stdlib_dep("syscall"))
+
+    direct_dependencies, extra_dependencies = await MultiGet(
+        Get(Targets, DependenciesRequest(target[Dependencies])),
+        Get(Targets, Addresses([*request.extra_dependencies, *extra_dependencies_addrs])),
+    )
 
     first_party_dep_import_path_targets = []
-    third_party_dep_import_path_targets = []
+    third_party_or_stdlib_dep_import_path_targets = []
     codegen_dep_import_path_targets = []
-    for dep in all_direct_dependencies:
+    for dep in direct_dependencies:
         if dep.has_field(GoPackageSourcesField):
             first_party_dep_import_path_targets.append(dep)
-        elif dep.has_field(GoThirdPartyPackageDependenciesField):
-            third_party_dep_import_path_targets.append(dep)
+        elif dep.has_field(GoImportPathField):
+            # Ensure marker packages C and unsafe do not show up as dependencies.
+            dep_import_path = dep[GoImportPathField].value
+            if dep_import_path not in {"builtin", "C", "unsafe"}:
```

Comment:
I get why C and builtin are skipped, but I'm not sure about unsafe. Does that not need to be built?

Reply:
All of the functions in `unsafe` are compiler intrinsics.

Reply:
So the `unsafe` package is provided by the compiler.